library/init/meta/lean/parser.lean

```diff
@@ -49,8 +52,26 @@ meta constant ident : parser name
 meta constant small_nat : parser nat
 /-- Check that the next token is `tk` and consume it. `tk` must be a registered token. -/
 meta constant tk (tk : string) : parser unit
-/-- Parse an unelaborated expression using the given right-binding power. -/
-protected meta constant pexpr (rbp := std.prec.max) : parser pexpr
+/-- Parse an unelaborated expression using the given right-binding power.
+     When `pat := tt`, the expression is parsed as a pattern, i.e. local
+     constants are not checked. -/
+protected meta constant pexpr (rbp := std.prec.max) (pat := ff) : parser pexpr
+
+
+/-- a variable to local scope -/
+meta constant add_local (v: expr) : parser unit
```

Comment:
```suggestion
meta constant add_local (v: expr) (is_variable := ff) : parser unit

```

Reply:
What would you do with this option?

Reply:
I was wondering if you should be able to simulate `variable x` from lean.  It's literally just an optional argument to `parser::add_local_expr`.

But don't feel obligated to add it.  There's a lot of API we could add.

Reply:
I see, yes it looks easy enough. I'll add it.

Reply:
Here are my thoughts on this feature: i won't add the flag but every time you call `add_local` it will be considered as a `variable`. If you use it inside of `with_local_scope`, it will be rolled back.